src/plonk_util.rs

```diff
@@ -244,6 +244,14 @@ pub fn sigma_polynomials<F: Field>(
         .collect()
 }
 
+pub(crate) fn polynomial_degree<F: Field>(
+    points: &[F],
+    fft_precomputation: &FftPrecomputation<F>,
+) -> usize {
+    let coeffs = ifft_with_precomputation_power_of_2(&points, fft_precomputation);
+    coeffs.iter().rev().skip_while(|c| c.is_zero()).count() - 1
+}
+
```

Comment:
I'm not sure how much it matters in practice, but this can give incorrect results for the degree. For example the polynomial `X^16n` will have degree 0 according to this function. I think we can add a check that the evaluation at a random point outside of `H` is consistent with the coefficients returned by the iFFT to be sure that the degree is less than `8n`.

Reply:
Whoops -- so my intention was to evaluate constraints over a larger domain, check the degree of the interpolant, and assert that it's within the max degree. But I was using the 8n subgroup as my "larger domain", and since the current max degree is also 8n, I wasn't catching anything.

I changed this to use a 16n subgroup, and a few of these tests (like Base4Sum's) are now failing as expected. Hopefully it makes sense now?